core/src/main/java/org/mule/config/DefaultMuleConfiguration.java

```diff
@@ -424,6 +427,26 @@ protected void validateXML() throws FatalException
         }
     }
 
+    /**
+     * Register a callback to be called when the logging configuration is changed.
+     * 
+     * @param listener the callback to execute
+     */
+    public static void registerLogConfigChangeListener(PropertyChangeListener listener)
+    {
+        ((LoggerContext) LogManager.getContext(false)).addPropertyChangeListener(listener);
```

Comment:
Logging (in general at least) in Mule is not directly dependent on Log4J, even if it is used by default.  So this either needs to use the logging abstraction, or handle the case where log4j is not being used.